src/test/java/de/tum/in/www1/artemis/StudentExamIntegrationTest.java

```diff
@@ -1321,6 +1323,7 @@ void testStudentExamSummaryAsStudentAfterPublishResults_dontFilter() throws Exce
             assertThat(exercise.getGradingCriteria()).isEmpty();
 
             if (exercise instanceof QuizExercise) {
```

Comment:
This allows to remove the cast below 
```suggestion
            if (exercise instanceof QuizExercise quizExercise) {
```